src/mame/video/namco_c355spr.cpp

```diff
@@ -423,19 +423,24 @@ void namco_c355spr_device::get_single_sprite(const u16 *pSource, c355_sprite *sp
 	}
 
 	if (num_rows == 0) num_rows = 0x10;
-	bool flipy = (vsize & 0x8000);
+	const bool flipy = (vsize & 0x8000);
 	vsize &= 0x3ff;
 	if (vsize == 0)
 	{
 		sprite_ptr->disable = true;
 		return;
 	}
 	u32 zoomy = (vsize << 16) / (num_rows * 16);
-	dy = (dy * zoomy + 0x8000) >> 16;
+	s32 dy_zoomed = ((dy & 0xff) * zoomy + 0x8000) >> 16;
+	if (dy & 0x100) dy_zoomed = -dy_zoomed;
```

Comment:
This effectively treats the value as sign/magnitude, and you get the “two zeroes” effect (0x000 and 0x100 are both treated as zero with it like this).  Can you confirm this is the correct?

Reply:
that appears to be the expected behaviour.